doc/examples/contextual_enhancement.py

```diff
@@ -0,0 +1,278 @@
+
+"""
+
+"""
+The enhancement is evaluated on the Stanford HARDI dataset 
+(150 orientations, b=2000s/mm^2) where Rician noise is added. Constrained 
+spherical deconvolution is used to model the fiber orientations.
+
+"""
+
+import numpy as np
+from dipy.data import fetch_stanford_hardi, read_stanford_hardi
+from dipy.sims.voxel import add_noise
+from dipy.core.gradients import gradient_table
+
+# Read data
+fetch_stanford_hardi()
+img, gtab = read_stanford_hardi()
+data = img.get_data()
+
+# Add Rician noise
+np.random.seed = 1234
+data_noisy = add_noise(data, 3, 500 , noise_type='rician')
+
+# Select a small part of it
+data_small = data[25:40, 65:80, 35:42]
+data_noisy_small = data_noisy[25:40, 65:80, 35:42]
+
+"""
+Perform a model fitting, in this case Constrained Spherical Deconvolution is used.
+"""
+
+# Perform CSD on the original data
+from dipy.reconst.csdeconv import auto_response
+from dipy.reconst.csdeconv import ConstrainedSphericalDeconvModel
+response, ratio = auto_response(gtab, data, roi_radius=10, fa_thr=0.7)
+csd_model_orig = ConstrainedSphericalDeconvModel(gtab, response)
+csd_fit_orig = csd_model_orig.fit(data_small)
+csd_shm_orig = csd_fit_orig.shm_coeff
+
+# Perform CSD on the original data + noise
+response, ratio = auto_response(gtab, data_noisy, roi_radius=10, fa_thr=0.7)
+csd_model_noisy = ConstrainedSphericalDeconvModel(gtab, response)
+csd_fit_noisy = csd_model_noisy.fit(data_noisy_small)
+csd_shm_noisy = csd_fit_noisy.shm_coeff
+
+"""
+Inspired by [Paulo_Eurographics_], a lookup-table is created, containing rotated 
+versions of the kernel :math:`P_t` sampled over a discrete set of orientations. 
+In order to ensure rotationally invariant processing, the discrete orientations 
+are required to be equally distributed over a sphere. By default, a sphere with 
+100 directions is used.
+"""
+
+from dipy.denoise.enhancement_kernel import EnhancementKernel
+from dipy.denoise.shift_twist_convolution import convolve
+
+# Create lookup table
+D33 = 1.0
+D44 = 0.02
+t = 1
+k = EnhancementKernel(D33, D44, t)
+
+"""
+Visualize the kernel
+"""
+
+from dipy.viz import fvtk
+from dipy.data import get_sphere
+from dipy.reconst.shm import sf_to_sh, sh_to_sf
+ren = fvtk.ren()
+
+# convolve kernel with delta spike
+spike = np.zeros((7, 7, 7, k.get_orientations().shape[0]), dtype=np.float64)
+spike[3, 3, 3, 0] = 1
+spike_shm_conv = convolve(sf_to_sh(spike, k.get_sphere(), sh_order=8), k, 
+                          sh_order=8, test_mode=True)
+
+sphere = get_sphere('symmetric724')
+spike_sf_conv = sh_to_sf(spike_shm_conv, sphere, sh_order=8)
+model_kernel = fvtk.sphere_funcs((spike_sf_conv*6)[3,:,:,:],
+                                  sphere, 
+                                  norm=False, 
+                                  radial_scale=True)
+fvtk.add(ren, model_kernel)
+fvtk.camera(ren, pos=(30, 0, 0), focal=(0, 0, 0), viewup=(0, 0, 1), verbose=False)
+fvtk.record(ren, out_path='kernel.png', size=(1024,1024))
```

Comment:
Reduce size to (900, 900). These figures go online. It's good strategy to reduce some bytes.
If you want you can use the new visualization API that allows you to handle the camera in a more robust way.
See the tutorials in the visualization section of the examples. For more ideas.
http://nipy.org/dipy/examples_index.html#visualization